### PR TITLE
FTP Operators new functions and bugfixes

### PR DESCRIPTION
**Pull request by joergboe**
Hi,

this pull requests brings new functions and bugfixes for the frp operators:
- Enhancements for issue #193 
- Enhancements in timeout handling
- Improvements in spl docs
- Bugfix for return status error in ftp command
- Bugfix for timeout handling, fused ftp operators and no async dns support (RHEL6)
- Bugfix for fused ftp operators for sftp protocoll

### CODE REVIEW

File: com.ibm.streamsx.inet/com.ibm.streamsx.inet.ftp/FTPCommand/FTPCommand.xml [hunk 1]
```diff
@@ -65,6 +65,37 @@ The command string and the command arguments are received from port 0.
             <prototype><![CDATA[<any T> T AsIs(T)]]></prototype>
           </function>
         </customOutputFunction>
+        <customOutputFunction>
+          <name>GetError</name>
+          <function>
+            <description>get the error number</description>
```
Comment (ddebrunner) [inline com.ibm.streamsx.inet/com.ibm.streamsx.inet.ftp/FTPCommand/FTPCommand.xml]: Uppercase 'G' in Get to match existing descriptions.

Comment (joergboe) [inline com.ibm.streamsx.inet/com.ibm.streamsx.inet.ftp/FTPCommand/FTPCommand.xml]: Ok I will change the descriptions in 'Get ...'

**Comment by joergboe** [comment]
I changed the function descriptions.

**Comment by joergboe** [comment]
I will close this pull request until I implement the changes discussed in issue #209

File: com.ibm.streamsx.inet/com.ibm.streamsx.inet.ftp/FTPCommand/FTPCommand.xml [hunk 2]
```diff
@@ -65,6 +65,33 @@ The command string and the command arguments are received from port 0.
...
+          <function>
+            <description>Get the error description</description>
+            <prototype><![CDATA[rstring ErrorText()]]></prototype>
+          </function>
+          <function>
+            <description>This function returns a string with the url including schema.</description>
+            <prototype><![CDATA[rstring Url()]]></prototype>
+          </function>
+          <function>
+            <description>Get the number of executed ftp commands(in case of rename only one is counted)</description>
```
Comment (ddebrunner) [inline com.ibm.streamsx.inet/com.ibm.streamsx.inet.ftp/FTPCommand/FTPCommand.xml]: Is this correct, as the description does not contain any mention of transfers?

File: com.ibm.streamsx.inet/com.ibm.streamsx.inet.ftp/FTPCommand/FTPCommand.xml [hunk 3]
```diff
@@ -65,6 +65,33 @@ The command string and the command arguments are received from port 0.
...
+            <prototype><![CDATA[uint32 NoTransfers()]]></prototype>
+          </function>
+          <function>
+            <description>Get the number of failed ftp commands</description>
```
Comment (ddebrunner) [inline com.ibm.streamsx.inet/com.ibm.streamsx.inet.ftp/FTPCommand/FTPCommand.xml]: Same comment. Is it transfers or commands?

**Comment by joergboe** [comment]
FTP Command operators executes commands. Thus there are no transfers mentioned.

**Pull request closed without merging**